library/modules/bigip_profile_persistence_src_addr.py

```diff
@@ -232,6 +255,32 @@ def parent(self):
         result = fq_name(self.partition, self._values['parent'])
         return result
 
+    @property
+    def mirror(self):
+        if self._values['mirror'] is None:
+            return None
+        result = flatten_boolean(self._values['mirror'])
+        if result is None:
+            return None
+        if result == 'yes':
+            return 'enabled'
+        return 'disabled'
+
+    @property
+    def mask(self):
```

Comment:
i would suggest finding a way to accept the CIDR and full mask format as it is better not to limit users options. Look here for some inspiration: https://github.com/F5Networks/f5-ansible/blob/devel/library/modules/bigip_apm_acl.py#L573

When you accept CIDR format just use the ipaddress lib functions to expand it to full format and send that to the API

Reply:
The problem is that of specifying whether the inputed value is IPv4 or IPv6. If it is something like '16' then it could be either. 

In that case, I'm forced to ask the user to tell me which version they're using. Also would need us to return the same.

Instead, this method is directly in line with the BIG-IP API itself.

I'll try to allow for both prefix and CIDR, but I don't see a lot of value in that.

Reply:
nah ok you convinced me, do something like this instead:

https://github.com/F5Networks/f5-ansible/blob/devel/library/modules/bigip_virtual_address.py#L388

Reply:
done!